### PR TITLE
Use vite for frontend development

### LINKED ISSUE #1796: Speed up frontend build and dev-server
I noticed the frontend build and especially the dev-server startup take quite some time.

To speed those up, I would propose migrating from webpack to [vite](https://github.com/vitejs/vite).
I am using vite in some projects and it really is quick :wink:.
The configuration is very similar to webpack (just simpler IMO).

I could open a PR if this is something you would consider.

### PR DESCRIPTION
**Pull request by lukashass**
Closes #1796 

## Summary

The scary amount of file changes :see_no_evil: is because I moved the static files from `dist` to `public` (this is the vite default which I find very useful as `dist` is now solely for build artifacts)

The most difficult part was building the iframe api. I did not find a proper way to include `iframe_api.ts` in the iframe while only using one vite instance. There are now two vite configs:
- `vite.config.ts` for building the frontend
- `iframe-api.vite.config.ts` with vite in [library mode](https://vitejs.dev/guide/build.html#library-mode) to output `iframe_api.js`

The iframe-api-typings are build using `tsc`.

## TODO

- [x] fix iframe-api
- [x] fix styling of loading screen (css loading)
- [x] migrate `build-typings`
- [x] fix `getEnvConfig`
- [x] fix routing when map url already present
- [x] ~async import `simple-peer`?~
- [x] remove webpack dependencies
- [x] add typecheck script
- [x] build iframe-api in CI
- [x] fix type error in `MapStore`
- [x] remove `iframe.html` workaround?
- [x] ~test if `LIVE_RELOAD` is still needed~ remove `LIVE_RELOAD`
- [x] update contributing docs

### CODE REVIEW

**Comment by lukashass** [comment]
> I don't think there is any need to async import simple-peer. I'm don't think we are doing it today by the way ;)

Thanks. I was just a bit confused because it was `require`d instead if `import`ed :blush: 

This should mainly be done now, except for some minor adjustments on the TODO list. I added a small summary to explain the main changes. @moufmouf Maybe you can have a look and check if I broke anything :see_no_evil:

File: front/src/Phaser/Game/GameMap.ts [hunk 1]
```diff
@@ -336,7 +336,9 @@ export class GameMap {
             if (!layer.visible) {
                 continue;
             }
-            if (layer.getTileAt(x, y)?.properties[GameMapProperties.COLLIDES]) {
+            // TODO: use optional chaining when https://github.com/DevExpress/testcafe-hammerhead/issues/2714 is fixed
+            const properties = layer.getTileAt(x, y)?.properties;
+            if (properties && properties[GameMapProperties.COLLIDES]) {
```
Comment (moufmouf) [inline front/src/Phaser/Game/GameMap.ts:341]: This is brilliant! Thanks!

**Comment by lukashass** [comment]
The e2e tests (except `variables.ts` not yet) are working without disabling the vite hot module replacement. I believe we can remove `LIVE_RELOAD` if the e2e tests were the only reason for it.

**Comment by lukashass** [comment]
https://github.com/thecodingmachine/workadventure/blob/0a705eadf9378bb7340b1aa04ac696c0b1ee29a2/front/src/Phaser/Items/Computer/computer.ts#L64

@moufmouf This comment is one of the last references to webpack. Is it important or can I remove it?

**Comment by moufmouf** [comment]
@lukashass you can definitely remove it! It's really old code regarding an early plug-in system that has been abandoned and will be superseded by the scripting API very soon.

**Comment by lukashass** [comment]
Is this the only issue solved by `iframe.html`? If that is the case `iframe.html` should no longer be necessary with vite.

```
Uncaught SyntaxError: The URL 'about:0.0.0.0:80/sockjs-node' is invalid
    at new SockJS (sockjs.js:679:1)
    at new SockJSClient (SockJSClient.js:43:1)
    at initSocket (socket.js:20:1)
    at Object../node_modules/webpack-dev-server/client/index.js?http://0.0.0.0:8080&sockPort=80 (index.js:176:1)
    at __webpack_require__ (bootstrap:24:1)
    at startup:5:1
    at startup:5:1
```

**Comment by moufmouf** [comment]
Hey @lukashass ,

I did some tests, deploying your PR on a continuous deployment environment, and I think there is an issue with simple-peer.

The Web-RTC connection we open with simple-peer does not open anymore.

I can see this error in one of the 2 browsers:

image

Any clue?

**Comment by moufmouf** [comment]
Test URL: https://play-vite.test.workadventu.re/_/global/maps-vite.test.workadventu.re/tests/jitsi_config.json

**Comment by moufmouf** [comment]
I think I get it!

Thanks for the work-around, it seems to be working great :+1:

**Comment by moufmouf** [comment]
I'm right now testing the E2E tests inside the CI pipeline.
For some weird reason, those are failing (but the E2E tests run locally are working).

Here is the error I see:

image

The problem seems to arise from the quill-delta-to-html library.
And of course, if I look at the package.json, I see: https://github.com/nozer/quill-delta-to-html/blob/master/package.json#L25-L26

I guess quill-delta-to-html will cause the same kind of troubles. I just can't understand why it is ok everywhere... except in the continuous integration environment.
Do you think we can apply the same kind of workaround?

**Comment by lukashass** [comment]
> Do you think we can apply the same kind of workaround?

This problem might be a bit different. Possibly related to https://github.com/nozer/quill-delta-to-html/issues/76, so maybe we can try [this suggestion](https://github.com/nozer/quill-delta-to-html/issues/76#issuecomment-663803383).

**Comment by lukashass** [comment]
> I'm right now testing the E2E tests inside the CI pipeline.
>  For some weird reason, those are failing (but the E2E tests run locally are working).

This should be solved by #1903. Waiting for it to be merged.

**Comment by moufmouf** [comment]
Ok, I believe we are ready for the big jump :D

Thanks a lot @lukashass  for this huge PR!

**Pull request merged**